packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -288,6 +291,191 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
+        const newEndRange = positionZoom + (endPosition - zoomIn * (endPosition * TEN_PERCENT));
+        return [newStartRange, newEndRange];
+    }
+
+    private zoomIn(startPos: number, endPos: number, percentMousePosition: number) {
+        const zoom = 1;
+        let startPosition = 0;
+        let endPosition = 0;
+        if (this.mouseIsDown) {
+            startPosition = startPos;
+            endPosition = endPos;
+        } else {
+            startPosition =   this.newRangeZoom(zoom, startPos, endPos)[0];
+            endPosition = this.newRangeZoom(zoom, startPos, endPos)[1];
+        }
+        this.updateRange(startPosition, endPosition, percentMousePosition);
+    }
+
+    private zoomOut(startPosition: number, endPosition: number, percentMousePosition: number) {
+        const zoom = -1;
+        this.updateRange( this.newRangeZoom(zoom, startPosition, endPosition)[0], this.newRangeZoom(zoom, startPosition, endPosition)[1],percentMousePosition);
+    }
+
+    private NewstartEndRangePanning(shift: number, percentRange: number): number[] {
+        const startRange = this.props.unitController.viewRange.start + (shift * percentRange);
+        const endRange = this.props.unitController.viewRange.end + (shift * percentRange);
+        const arr = [startRange, endRange];
+        return arr;
+    }
+
+    private updatePositionAfterPann(right: number, distance: number){
+        this.positionZoomCenter = this.positionZoomCenter + right * distance;
+        this.positionZoomMouse = this.positionZoomMouse + right * distance;
+    }
+
+    private shift(right: number, distance: number) {
+        const percentRange = this.props.unitController.viewRangeLength * TEN_PERCENT;
+        if (this.NewstartEndRangePanning(right, percentRange)[0] < 0) {
+            this.updatePositionAfterPann(right,distance);
+            this.props.unitController.viewRange = {
+                start: 0,
+                end: this.props.unitController.viewRangeLength
+            };
+        } else if (this.NewstartEndRangePanning(right, percentRange)[1] > this.props.unitController.absoluteRange) {
+            this.updatePositionAfterPann(right,distance);
+            this.props.unitController.viewRange = {
+                start: this.props.unitController.absoluteRange - this.props.unitController.viewRangeLength,
+                end: this.props.unitController.absoluteRange
+            };
+        } else {
+            this.updatePositionAfterPann(right, percentRange);
+            this.props.unitController.viewRange = {
+                start: this.NewstartEndRangePanning(right, percentRange)[0],
+                end: this.NewstartEndRangePanning(right, percentRange)[1]
+            };
+        }
+    }
+
+    private shiftLeft() {
+        const numberLeft = -1;
+        const distance = ( this.props.unitController.viewRange.start);
+        this.shift(numberLeft, distance);
```

Comment:
Would be cleaner to pass a boolean and these two values could be calculated within the called method instead?